src/test/store/zipped-profiles.test.js

```diff
@@ -0,0 +1,178 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import {
+  formatZipFileTable,
+  storeWithZipFile,
+} from '../fixtures/profiles/zip-file';
+import { procureInitialInterestingExpandedNodes } from '../../profile-logic/zip-files';
+import * as ProfileViewSelectors from '../../reducers/profile-view';
+import * as ZippedProfilesSelectors from '../../reducers/zipped-profiles';
+import createStore from '../../create-store';
+import { ensureExists } from '../../utils/flow';
+
+import JSZip from 'jszip';
+
+import * as ZippedProfilesActions from '../../actions/zipped-profiles';
+import * as ReceiveProfileActions from '../../actions/receive-profile';
+
+describe('reducer zipFileState', function() {
+  it('can store the zip file in the reducer', async function() {
+    const { zippedProfiles } = await storeWithZipFile();
+    expect(zippedProfiles).toBe(zippedProfiles);
+  });
+
+  it('can load a profile from the zip file', async function() {
+    const { dispatch, getState } = await storeWithZipFile([
+      'foo/bar/profile1.json',
+      'foo/profile2.json',
+      'baz/profile3.json',
+    ]);
+    expect(ProfileViewSelectors.getProfileOrNull(getState())).toEqual(null);
+
+    await dispatch(
+      ZippedProfilesActions.viewProfileFromPathInZipFile(
+        'foo/bar/profile1.json'
+      )
+    );
+
+    expect(ZippedProfilesSelectors.getZipFileState(getState()).phase).toBe(
+      'VIEW_PROFILE_IN_ZIP_FILE'
+    );
+    const profile1 = ProfileViewSelectors.getProfile(getState());
+
+    expect(profile1).toBeTruthy();
+  });
+
+  it('will fail when trying to load an invalid profile', async function() {
+    const store = createStore();
+    const { getState, dispatch } = store;
+    const zip = new JSZip();
+    zip.file('not-a-profile.json', 'not a profile');
+    dispatch(ReceiveProfileActions.receiveZipFile(zip));
+
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    await dispatch(
+      ZippedProfilesActions.viewProfileFromPathInZipFile('not-a-profile.json')
+    );
+
+    expect(ZippedProfilesSelectors.getZipFileState(getState()).phase).toEqual(
+      'FAILED_TO_PROCESS_PROFILE_FROM_ZIP_FILE'
+    );
+    // console error was called.
+    expect(console.error.mock.calls.length >= 1).toEqual(true);
+    expect(console.error.mock.calls).toMatchSnapshot();
+
+    console.error.mockRestore();
```

Comment:
This line isn't even needed now, thanks to:
https://github.com/devtools-html/perf.html/blob/0161ae4ce1362a5e79082c3fdd8665dc96b71a5d/package.json#L136